test/unit/puppetlabs/services/request_handler/request_handler_core_test.clj

```diff
@@ -311,6 +311,17 @@
                      custom-vcs
                      tk-scheduler/scheduler-service]]
        (jruby-bootstrap/with-puppetserver-running-with-services-and-mock-jrubies
+        "MOCK.TODO: This test is making a catalog request, in order
+        to trigger the code-id handling.  With the current jruby mocking that
+        means that we are artificially returning a 200 for that catalog request,
+        but this test passes because it doesn't make any assertions on the body.
+        The test description probably needs updating too... it seems like the
+        goal of this test is to assert that the jruby is borrowed before the
+        code id command is run, so that we know that a lock can't be acquired
+        (and new code deployed) while we're running the code id command.  If that's
+        the case, we should explain it more clearly, and it's probably OK to
```

Comment:
That all makes sense to me.  I agree that we're validating that the code id command involves reserving a JRuby specifically so that the pool cannot be locked (and therefore possibly code being updated) while the code id is being computed.  I'll make an attempt to add an extra comment about the purpose of the test in the comment under the `testing` block above.